### ISSUE TITLE
handle-done-callback: allow passing done as callback

### ISSUE THREAD

**Issue by shaharmor** [lo1tuma/eslint-plugin-mocha#119]
this should not cause an error:

```js
it('should not cause an error', function(done) {
  var obj = {someFunc: done};
  somethingThatCallsSomeFuncOnObj(obj);
});
```

should it?

**Comment by lo1tuma** [comment]
Yes, I think you are right.
The current implementation explicitly checks for a `CallExpression` of `done` or its direct parent, so passing `done` to a function should work but not if it is nested in other expressions like an object literal.
Maybe we should just relax the rule and treat any read of the done callback as "handled".

What do you think @jfmengels?

**Comment by jfmengels** [comment]
I agree with you, the code excerpt should not cause an error.

We do lose the "assurance" that the function is at least called at some point in the code, but that did not make sure that the callback was correctly triggered anyway. Just checking the presence of an `Identifier` would be fine IMO.

**Comment by edg2s** [comment]
Wouldn't that then just be a very specific version of `no-unused-vars`, which is in eslint:recommend?
